rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -83,520 +90,390 @@ fn oversized_reject_message_is_truncated() {
     );
 }
 
+/// Tests that inducting a loopback stream where `signals_end` < `end` panics (this must be equal
+/// for the loopback stream, because it is its own reverse stream).
 #[test]
 #[should_panic(
     expected = "Invalid message indices in stream slice from subnet g24bn-xymaa-aaaaa-aaaap-yai: messages begin (21) != stream signals_end (22)"
 )]
-fn induct_loopback_stream_with_signals_panics() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut state, _) = new_fixture(&log);
-
-        // A loopback stream containing 2 messages and 1 signal.
-        let loopback_stream = generate_loopback_stream(StreamConfig {
-            messages_begin: 21,
-            message_count: 2,
+fn induct_loopback_stream_with_signals_end_before_messages_end_panics() {
+    let (stream_handler, state, _) =
+        with_local_test_setup(btreemap![LOCAL_SUBNET => StreamConfigX {
+            begin: 21,
+            messages: vec![*LOOPBACK_REQUEST_CONFIG; 2],
             signals_end: 22,
-            reject_signals: None,
-        });
-        state.with_streams(btreemap![LOCAL_SUBNET => loopback_stream]);
+            ..StreamConfigX::default()
+        }]);
 
-        stream_handler.induct_loopback_stream(state, &mut 1_000_000_000);
-    });
+    stream_handler.induct_loopback_stream(state, &mut (i64::MAX / 2));
 }
 
+/// Tests that inducting a loopback stream where `signals_end` < `begin` panics (this must be equal
+/// for the loopback stream, because it is its own reverse stream).
 #[test]
 #[should_panic(
     expected = "Invalid message indices in stream slice from subnet g24bn-xymaa-aaaaa-aaaap-yai: messages begin (21) != stream signals_end (20)"
 )]
 fn induct_loopback_stream_signals_end_before_messages_begin_panics() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut state, _) = new_fixture(&log);
-
-        // A loopback stream with signals().end() != messages().begin().
-        let loopback_stream = generate_loopback_stream(StreamConfig {
-            messages_begin: 21,
-            message_count: 2,
+    let (stream_handler, state, _) =
+        with_local_test_setup(btreemap![LOCAL_SUBNET => StreamConfigX {
+            begin: 21,
+            messages: vec![*LOOPBACK_REQUEST_CONFIG; 2],
             signals_end: 20,
-            reject_signals: None,
-        });
-        state.with_streams(btreemap![LOCAL_SUBNET => loopback_stream]);
+            ..StreamConfigX::default()
+        }]);
 
-        stream_handler.induct_loopback_stream(state, &mut 1_000_000_000);
-    });
+    stream_handler.induct_loopback_stream(state, &mut (i64::MAX / 2));
```

Comment:
Only just noticed this: `with_test_replica_logger()` expects all of the work to happen within the closure; calling `induct_loopback_stream()` (or whatever else, in other tests) outside `with_test_replica_logger()` will result in any log messages produced by the test code never being printed.

I.e. you need to refactor your `with_*_test_setup_*()` functions to also take the test body as a closure and execute it within `with_test_replica_logger()`. Something like:

```rust
with_local_test_setup(
    btreemap![LOCAL_SUBNET => StreamConfigX {
        ...
    }],
    |stream_handler, state, _slies, _log| {
        stream_handler.induct_loopback_stream(state, &mut (i64::MAX / 2));
    }
)
```

Reply:
This is what I wanted to do at first but then didn't because the formatter indents it more than I was hoping for.  After playing around with it a bit, I am fairly certain that you can't prevent this unless perhaps with a macro. But I guess this extra indentation isn't so bad.

Reply:
It actually turned out ok. I was expecting to make sky scrappers out of everything 🫢